### PR TITLE
Test to verify Ceph configuration values are accessible via cephadm-ansible

### PR DESCRIPTION
**Pull request by s-vipin**
Automation of the issues - [BZ#2213766](https://bugzilla.redhat.com/show_bug.cgi?id=2213766)

Polarion: [CEPH-83602686](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83602686) 

Steps:
    1. Set mgr/dashboard/node_name/server_addr configuration
       of mon daemon to IP address using `ceph config set` command.
    2. Retrieve configuration value using `ceph config get` command.
    3. Execute `cephadm-ansible` playbook to retrieve configuration
       value.
    4. Validate `cephadm-ansible` playbook output contains configuration value
       from step #2.

Pass log: http://magna002.ceph.redhat.com/ceph-qe-logs/vips/logs_24_12_04_16:52:31/ 

# Description 

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: ceph/rados/core_workflows.py [hunk 1]
```diff
@@ -4615,3 +4615,68 @@ def get_rados_df(self, pool_name: str = None):
         out = self.run_ceph_command(cmd=_cmd, client_exec=True)
 
         return out["pools"][0] if pool_name else out
+
+    def set_configuration(self, who: str, param: str, value: str) -> int:
```
Comment (pdhiran) [inline ceph/rados/core_workflows.py:4619]: Module for this already exists. please refer to `set_config()` in `tests/rados/monitor_configurations.py`

File: ceph/rados/core_workflows.py [hunk 2]
```diff
@@ -4615,3 +4615,68 @@ def get_rados_df(self, pool_name: str = None):
...
+        actual_value = self.get_configuration(who=who, param=param)
+        log.info(f"actual value: {actual_value}")
+        log.info(f"value: {value}")
+        if actual_value == value:
+            log.info(f"Configuration parameter {param} is set with {value} on {who}")
+            return True
+        else:
+            return False
+
+    def get_configuration(self, who: str, param: str) -> int:
```
Comment (pdhiran) [inline ceph/rados/core_workflows.py:4646]: Module for this already exists. please refer to `get_config()` in `tests/rados/monitor_configurations.py`

File: ceph/rados/core_workflows.py [hunk 3]
```diff
@@ -4615,3 +4615,68 @@ def get_rados_df(self, pool_name: str = None):
...
+        Usage:
+            rados_obj.set_configuration(who="osd", param="osd_mclock_override_recovery_settings", value="true")
+            rados_obj.set_configuration(who="mgr", param="mgr/dashboard/ceph1/server_addr", value="10.74.250.90")
+        """
+
+        log.info(f"Getting {who} configuration parameter {param}")
+        cmd = f"ceph config get {who} {param}"
+        return self.client.exec_command(cmd=cmd)[0].strip()
+
+    def remove_configuration(self, who: str, param: str) -> int:
```
Comment (pdhiran) [inline ceph/rados/core_workflows.py:4665]: Module for this already exists. please refer to `remove_config()` in `tests/rados/monitor_configurations.py`

File: suites/quincy/rados/tier-2_rados_test-bugfixes.yaml [hunk 4]
```diff
@@ -204,3 +205,9 @@ tests:
   #     module: test_mon_connection_scores.py
   #     polarion-id: CEPH-83602911
   #     comments: Active bug 2151501
+
+  - test:
+      name: Verification of configuration values via cephadm-ansible module
```
Comment (pdhiran) [inline suites/quincy/rados/tier-2_rados_test-bugfixes.yaml:210]: please shorten the name of test case.

File: tests/rados/test_bz_2213766.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
+from ceph.ceph_admin import CephAdmin
```
Comment (pdhiran) [inline tests/rados/test_bz_2213766.py:1]: Please add docstrings for the test file.

File: tests/rados/test_bz_2213766.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+EOF"""
+        rc = installer_node.exec_command(cmd=cmd, long_running=True, sudo=True)
+
+        if rc != 0:
+            raise AssertionError("ansible-playbook failed to execute")
+
+        cmd = f"cd {ansible_dir} ; ansible-playbook -v -i {inventory} {ansible_dir}/test_bz_2213766.yaml"
+        out, err = installer_node.exec_command(cmd=cmd, sudo=True)
+        if not err:
+            raise AssertionError("ansible-playbook failed to execute")
```
Comment (pdhiran) [inline tests/rados/test_bz_2213766.py:95]: is the assertion condition correct here? if Yes, please add some comments in the file as to why not having error lines is considered failure condition.

Also, if output is not too huge, can we print the console output obtained from running the playbook?

**Review by pdhiran — COMMENTED** [review]
looks good overall.

**Comment by pdhiran** [comment]
polarion test approved : https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83602686

File: ceph/rados/core_workflows.py (hunk 1 above)
Comment (s-vipin) [reply ceph/rados/core_workflows.py]: Reused existing function set_config()
` config_obj.set_config(
                section=config_section,
                name=config_name,
                value=config_value,
            )`
https://github.com/red-hat-storage/cephci/blob/262205193a7497c6e25014f51a479518e7a1ba85/tests/rados/test_bz_2213766.py#L69-L73

File: ceph/rados/core_workflows.py (hunk 2 above)
Comment (s-vipin) [reply ceph/rados/core_workflows.py]: Updated code to reuse function get_config() from tests/rados/monitor_configurations.py

https://github.com/red-hat-storage/cephci/blob/262205193a7497c6e25014f51a479518e7a1ba85/tests/rados/test_bz_2213766.py#L85-L88

File: ceph/rados/core_workflows.py (hunk 3 above)
Comment (s-vipin) [reply ceph/rados/core_workflows.py]: Updated to reuse existing remove_config() 

https://github.com/red-hat-storage/cephci/blob/262205193a7497c6e25014f51a479518e7a1ba85/tests/rados/test_bz_2213766.py#L149-L152

File: suites/quincy/rados/tier-2_rados_test-bugfixes.yaml (hunk 4 above)
Comment (s-vipin) [reply suites/quincy/rados/tier-2_rados_test-bugfixes.yaml]: Updated test name 
https://github.com/red-hat-storage/cephci/blob/262205193a7497c6e25014f51a479518e7a1ba85/suites/quincy/rados/tier-2_rados_test-bugfixes.yaml#L209-L213

File: tests/rados/test_bz_2213766.py (hunk 5 above)
Comment (s-vipin) [reply tests/rados/test_bz_2213766.py:8]: Added doc string

https://github.com/red-hat-storage/cephci/blob/262205193a7497c6e25014f51a479518e7a1ba85/tests/rados/test_bz_2213766.py#L1-L6

**Comment by pdhiran** [comment]
Waiting for pass log post review changes

**Comment by s-vipin** [comment]
Pass logs for squid along with entire suite: http://magna002.ceph.redhat.com/ceph-qe-logs/vips/logs_ceph_adm_squid_24_12_12_11_25_09/

**Review by pdhiran — APPROVED** [review]
Looks good

**Pull request merged**